File: backend/src/main/java/com/poeflipfinder/backend/usecase/runingestioncatchup/CatchupCapPolicy.java
package com.poeflipfinder.backend.usecase.runingestioncatchup;

import java.time.Duration;

/**
 * The two tuning knobs for one bounded catch-up walk (docs/ARCHITECTURE.md §
 * Currency Exchange Ingestion), bundled since they're always supplied
 * together from application.yml.
 */
public record CatchupCapPolicy(int maxHoursPerCall, Duration firstRunLookback) {
}

File: app/lib/poe_flip_finder/catchup_cap_policy.ex
defmodule PoeFlipFinder.CatchupCapPolicy do
  @moduledoc """
  The two tuning knobs for one bounded catch-up walk
  (docs/ARCHITECTURE.md § Currency Exchange Ingestion), bundled since
  they're always supplied together from application config.
  """

  @enforce_keys [:max_hours_per_call, :first_run_lookback_hours]
  defstruct [:max_hours_per_call, :first_run_lookback_hours]

  @type t :: %__MODULE__{
          max_hours_per_call: pos_integer(),
          first_run_lookback_hours: pos_integer()
        }
end
